### ISSUE TITLE
Reduce duplicate code across plugins using notify popup

### ISSUE THREAD

**Issue by kirsty-hames** [adaptlearning/adapt-contrib-vanilla#443]
### Subject of the enhancement
Currently the default style for Notify icon buttons is consistent across all plugins however we've duplicated the styling code across the various plugin LESS files. 

See Hotgraphic close button in comparison to core Notify close button. The CSS being applied is the same but attached to different selectors.

<img width="1911" alt="notify_close_btn" src="https://github.com/adaptlearning/adapt-contrib-vanilla/assets/7045330/6295297e-c4df-4111-a1a0-9ca538c0ab18">

See references to duplicate styles below:
- Notify targets [`.notify__btn-icon`](https://github.com/adaptlearning/adapt-contrib-vanilla/blob/142b4fde60d638e280d04d691c63fbe62bd5abfe/less/core/notify.less#LL74C1-L86C4)
- Tutor targets [`.tutor__btn-icon`](https://github.com/adaptlearning/adapt-contrib-vanilla/blob/142b4fde60d638e280d04d691c63fbe62bd5abfe/less/plugins/adapt-contrib-tutor/tutor.less#L20)
- Hotgraphic targets [`.hotgraphic-popup__controls` and `.hotgraphic-popup__close`](https://github.com/adaptlearning/adapt-contrib-vanilla/blob/142b4fde60d638e280d04d691c63fbe62bd5abfe/less/plugins/adapt-contrib-hotgraphic/hotgraphicPopup.less#LL9C1-L22C4)

### Expected behaviour
The generic Notify button icon selector `.notify__btn-icon` is only being used in the [notifyPopup](https://github.com/adaptlearning/adapt-contrib-core/blob/24643de7fd2c62adca3ed2bff19ef48be460d73d/templates/notifyPopup.hbs#L98) template.

I'd expect `.notify__btn-icon` to be used for all Notify icon buttons. This means the default style is set in one place but we still have plugin specific classes to target should you want to style these different - although there's good reason for keeping UI button styling consistent. 

### Next steps
- Raise issues and PRs for the plugins where the additional selector `.notify__btn-icon` is applicable.
- Only once plugin PRs are merged should the plugin styles be removed from Vanilla. As we're removing styles that require the compatible plugin? Is this a breaking change for Vanilla or the relevant plugin? Or both?

**Comment by StuartNicholls** [comment]
@kirsty-hames, as a general question, what are the advantages of having four separate classes/selectors for the close button? As apposed to say, one class/selector called close-btn. The selector can target the icon with a mixin for example. 

The seems to be some confusion looking at how this has been implemented generally speaking. Across plug-ins, but even in the theme for example https://github.com/adaptlearning/adapt-contrib-vanilla/blob/142b4fde60d638e280d04d691c63fbe62bd5abfe/less/core/notify.less#L79 - looking at the I'd expect the radius to apply to the button, rather then in the selector than deals with the icon. 

I'm just wondering if it might be better to create a new selector, call it close-btn, then leave those specific selectors then tidy up the less. Thoughts?

**Comment by kirsty-hames** [comment]
> @kirsty-hames, as a general question, what are the advantages of having four separate classes/selectors for the close button? As apposed to say, one class/selector called close-btn. The selector can target the icon with a mixin for example.

I completely agree. On the off chance you did want to do different styles you can still target a parent selector for a particular plugin.  Whether we touch the plugin selectors or not a single mixin in the theme would reduce the need for duplicate CSS and resolve the issue raised.

> There seems to be some confusion looking at how this has been implemented generally speaking - across plug-ins, but even in the theme for example
> 
> https://github.com/adaptlearning/adapt-contrib-vanilla/blob/142b4fde60d638e280d04d691c63fbe62bd5abfe/less/core/notify.less#L79
> 
> Looking at the naming logic for the classes, I'd expect the radius to apply to the button, rather than in the selector than deals with the icon.

I'd also agree. I think this needs to be looked at from a wider UI perspective though (same issue applies to Drawer buttons etc) so I think this can be a separate issue/task. For the purpose of this ticket, attaching the current styles to mixin is a starting point.

> I'm just wondering if it might be better to create a new selector, call it close-btn, then leave those specific selectors (for customisations) then tidy up the less. Thoughts?

Yep agree again, leaving the selectors as is avoids any breaking changes/compatibility issues across Vanilla and plugins so I prefer this approach. Thanks for raising. We just need to rethink the mixin name as I raised this issue for Notify icon buttons not just close so would apply to the controls too (back/next) but your suggestion of a mixin is a good way to go for this.

**Comment by StuartNicholls** [comment]
@kirsty-hames , what I meant was re. the new selector, was create a new selector and add the class to the DOM so, call it adapt-close-btn , but perhaps thats out of scope here. I suggested that as you were suggesting adding `notify__btn-icon`, to hotgrid etc. this is slightly confusing as its BEM style class but used like utility class if we add it here.

So for now, yeah lets stay away from adding that class to the DOM in other plug-ins and perhaps add in XXX_btn-icon (so hotgrid_btn-icon in the case of hotgrid) so it (sort of) follows the same BEM naming. And use a mixin like we said above. 

All the other stuff about utility classes and rationing the ones already in use can get raised in another ticket. 

What do you think?

**Comment by kirsty-hames** [comment]
> @kirsty-hames , what I meant was re. the new selector, was create a new selector and add the class to the DOM so, call it adapt-close-btn , but perhaps thats out of scope here. I suggested that as you were suggesting adding `notify__btn-icon`, to hotgrid etc. this is slightly confusing as its BEM style class but used like utility class if we add it here.
> 
> So for now, yeah lets stay away from adding that class to the DOM in other plug-ins and perhaps if we need to add in XXX_btn-icon (so hotgrid_btn-icon in the case of hotgrid) so it (sort of) follows the same BEM naming. And use a mixin like we said above. Otherwise just a mixin.
> 
> Added a comment here:
> 
> [cgkineo/adapt-hotgrid#127 (comment)](https://github.com/cgkineo/adapt-hotgrid/pull/127#discussion_r1205528184)
> 
> All the other stuff about utility classes and rationalising the ones already in use can get raised in another ticket.
> 
> What do you think?

Lets just go with the mixin. I'd say adding/changing classes is out of scope for this issue. The mixin alone resolves the issue of duplicate styles.
